File: OCP/Solution/MarsRoverKata/Obstacle.cs
namespace MarsRoverKata
{
    public class Obstacle
    {
        public Point Location { get; protected set; }

        public Obstacle(Point location, bool isDestructable)
        {
            Location = location;
            IsDestructable = isDestructable;
        }

        public bool IsDestructable { get; private set; }
    }
}

File: OCP/Java_Exercise/MarsRoverKata/src/main/java/MarsRoverKata/Obstacle.java
package MarsRoverKata;

public class Obstacle implements IObstacle {
    private Point location;

    protected void setLocation(Point location) {
        this.location = location;
    }

    public Point getLocation() {
        return location;
    }

    public Obstacle(Point location) {
        this.location = location;
    }

    public boolean isDestructable() {
        return true;
    }
}
